modules/admin/src/test/java/org/motechproject/admin/it/AdminBundleIT.java

```diff
@@ -114,6 +129,56 @@ public void testMessageController() throws IOException, InterruptedException {
         assertTrue("No messages listed", json.size() > 0);
     }
 
+    @Test
+    public void testUploadBundleFromRepository() throws IOException, InterruptedException {
+        boolean isUploadSuccessful = uploadBundle("Repository", "org.motechproject:cms-lite:LATEST", null, "on");
+
+        assertTrue(isUploadSuccessful);
+    }
+
+    @Test
+    public void testUploadBundleFromFile() throws IOException, InterruptedException {
+        File file = new File("src/test/resources/motech-upload-test-bundle.jar");
+
+        boolean isUploadSuccessful = uploadBundle("File", null, file, "on");
+
+        assertTrue(isUploadSuccessful);
+    }
+
+    private boolean uploadBundle(String moduleSource, String moduleId, File bundleFile, String startBundle) throws IOException, InterruptedException {
```

Comment:
General note: instead of this method returning false, it should invoke fail() method, with meaningful message, what went wrong. At the moment, if this test fails, all that will be shown in the logs is that the assertion isUploadSuccessful failed.